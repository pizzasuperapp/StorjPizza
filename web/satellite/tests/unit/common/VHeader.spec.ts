// Copyright (C) 2019 Storj Labs, Inc.
// See LICENSE for copying information.

import HeaderComponent from '@/components/common/VHeader.vue';

import { mount, shallowMount } from '@vue/test-utils';

describe('HeaderComponent.vue', () => {
    it('renders correctly', () => {
        const wrapper = shallowMount<HeaderComponent>(HeaderComponent);

        expect(wrapper).toMatchSnapshot();
    });

    it('renders correctly with default props', () => {
        const wrapper = mount<HeaderComponent>(HeaderComponent);
        expect(wrapper.vm.$props.placeholder).toMatch('');
    });

    it('function clearSearch works correctly', () => {
        const search = jest.fn();

        const wrapper = mount<HeaderComponent>(HeaderComponent, {
            propsData: {
                search: search,
            }
        });
        wrapper.vm.clearSearch();
        expect(search).toHaveBeenCalledTimes(1);
    });
});
