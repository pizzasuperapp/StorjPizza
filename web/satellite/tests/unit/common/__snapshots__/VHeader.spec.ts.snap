// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent.vue renders correctly 1`] = `
<div class="header-container">
  <div class="header-container__buttons-area"></div>
  <div>
    <vsearch-stub placeholder="" search="[Function]"></vsearch-stub>
  </div>
  <!---->
</div>
`;
